### PR TITLE
Update CSS to also work with lessons

### PR DESCRIPTION
**Pull request by jpallen**
With the addition of a few HTML classes at the top level, the lessons template
should now work with this CSS, so long as it includes swc.css,
swc-workshop-and-lesson.css and lesson.css. The important thing is that there
aren't any conflicts between classes in site and lessons anymore, so how the
CSS files are grouped and delivered is easy to change. Perhaps everything
should be shrunk into just swc.css for use everywhere?

This has a small effect on site: h1.title in site is used to add an underline,
but only actually used in the badges pages. The CSS seems broken on these pages
anyway, so I've removed this class in the CSS since it conflicts with h1.title
in lessons.

Another point to note is that I've changed the global selectors for
h1:first-child and h2:first-child to only affect .objective, .callout and
.challenge blocks, since the global selector used for lessons conflicted with
site. However, the selector now only picks up h2:first-child, not h1 since this
seemed to be all that was used for headers in these blocks.  It's easy to also
include h1 if needed though.

Finally, there is a .keypoints selector in the current lesson CSS, but I see no
mention of that in the blockquote2div.py filter in the lesson template. There
is a .prereq class however in blockquote2div.py which currently has no CSS.
Perhaps .keypoints -> .prereq? Or at least we can remove .keypoints?

I've only checked this thoroughly with shell-novice so far, so there may be
issues with the other lessons that I haven't audited.

### CODE REVIEW

**Comment by jpallen** [comment]
The corresponding changes needed in the lesson template are at swcarpentry/lesson-template#140

File: css/lesson.css [hunk 1]
```diff
@@ -1,47 +1,126 @@
...
 
-/* Continuation paragraph. */
-p.continue {}
+/* Database tables do _not_ have double borders */
+table.outlined {
+    border-collapse: collapse;
+}
+    
+/* Printing */
+@media print {
```
Comment (r-gaia-cs) [inline css/lesson.css]: The CSS for `print` was design targeting only the lessons. It need to be improved **in another pull request**.

**Comment by r-gaia-cs** [comment]
> This has a small effect on site: h1.title in site is used to add an underline,
> but only actually used in the badges pages. The CSS seems broken on these pages
> anyway, so I've removed this class in the CSS since it conflicts with h1.title
> in lessons.

@jpallen Thanks for the awesome work. Could you merge https://github.com/jpallen/site/pull/1?

> Another point to note is that I've changed the global selectors for
> h1:first-child and h2:first-child to only affect .objective, .callout and
> .challenge blocks, since the global selector used for lessons conflicted with
> site.

Thanks.

> However, the selector now only picks up h2:first-child, not h1 since this
> seemed to be all that was used for headers in these blocks. It's easy to also
> include h1 if needed though.

Our template says to only use `h2` so this will be OK for now.

> Finally, there is a .keypoints selector in the current lesson CSS, but I see no
> mention of that in the blockquote2div.py filter in the lesson template.

`.keypoints` was used on https://github.com/swcarpentry/bc. We must preserve it since we will keep the v5.2 on our site, see http://software-carpentry.org/lessons.html.

> There is a .prereq class however in blockquote2div.py which currently has no CSS.
> Perhaps .keypoints -> .prereq?

`.keypoints` is more or less a duplicate of `.objectives`. Since we are going to try keep small lessons we will not need it anymore.

> Or at least we can remove .keypoints?

We should preserve `.keypoints` at least for now. **See previous comment.**

File: css/lesson.css (hunk 1 above)
Comment (jpallen) [inline css/lesson.css]: It looks like most of the elements affected are only in lessons anyway. The exceptions are h1,h2,h3,h4,h5,h6,p,ul,ol,li,pre,code, which may also affect other things globally in site and workshop.

Presumably the print styles for these elements in lesson and workshop should be the same anyway? Is their a maintained print style for site? It seems less important than the print style for lessons and workshops.

Do you want me to remove or fix this in some way before merging the PR?

**Comment by jpallen** [comment]
> .keypoints was used on https://github.com/swcarpentry/bc. We must preserve it since we will keep the v5.2 on our site, see http://software-carpentry.org/lessons.html.

It looks like the linked lessons are namespaced under /v5 on the site, and link to CSS that is also namespaced under /v5. Does this CSS come from the 'bc' repo as well? In which case these changes shouldn't affect it anyway?

**Comment by gvwilson** [comment]
I apologize for the complexity - it's grown over a couple of years, and 
not in any sensible way.  Looking at the web site, it looks like the 
material in /v5 (which comes from the 'bc' repository) is referencing 
/v5/css/*.css, but the stuff in the main directory (which comes from 
'site') is referencing ./css (which also comes from 'site').

**Comment by gvwilson** [comment]
@r-gaia-cs have you checked this?

**Comment by r-gaia-cs** [comment]
@jpallen Sorry for the delay. I was trying to find something equivalent to Jinja's powerful [template inheritance](http://jinja.pocoo.org/docs/dev/templates/#template-inheritance) for Jekyll to avoid the duplicate code at `_layouts` but I didn't find.

Merging now.

**Comment by jpallen** [comment]
@r-gaia-cs No problem, I'm struggling to keep track of all of these CSS PRs, and I wrote some of them! Hopefully things will be easier once it's all consolidated :) Thanks for merging! I guess this now needs to make it into swcarpentry/styles as well?

**Comment by r-gaia-cs** [comment]
> I guess this now needs to make it into swcarpentry/styles as well?

I will try to address it.

**Comment by gvwilson** [comment]
May I suggest:

1. We create a styles-test repo.

2. We put a bunch of HTML files in there that are copied from various 
other repos.

3. People test all their CSS changes there.

4. When things are done, they manually copy over to the 'styles' repo 
and submit there.

So there's only one canonical set of CSS (in 'styles', or in PRs against 
'styles'), and we can view changes simultaneously across 
lesson-template, workshop-template, and site by proxy.

**Pull request merged**